src/libs/Network.js

```diff
@@ -29,6 +30,54 @@ Onyx.connect({
     callback: val => email = val ? val.email : null,
 });
 
+/**
+ * Checks to see if a request should be made.
+ *
+ * @param {Object} request
+ * @param {Object} request.data
+ * @param {Boolean} request.data.forceNetworkRequest
+ * @return {Boolean}
+ */
+function shouldMakeRequest(request) {
+    // These requests are always made even when the queue is paused
+    if (request.data.forceNetworkRequest === true) {
+        return true;
+    }
+
+    if (isQueuePaused) {
+        return false;
+    }
+
+    return true;
+}
+
+/**
+ * Checks to see if a request should be retried when the queue is "paused".
+ *
+ * @param {Object} request
+ * @param {String} request.command
+ * @param {Object} request.data
+ * @param {Boolean} request.data.doNotRetry
+ * @param {String} [request.data.returnValueList]
+ * @return {Boolean}
+ */
+function shouldRetryRequest(request) {
+    const doNotRetry = lodashGet(request, 'data.doNotRetry', false);
+    const logParams = {command: request.command, doNotRetry, isQueuePaused};
+    const returnValueList = lodashGet(request, 'data.returnValueList');
+    if (returnValueList) {
+        logParams.returnValueList = returnValueList;
+    }
+
+    if (doNotRetry) {
+        console.debug('Skipping request that should not be re-tried: ', logParams);
```

Comment:
Yeah, that's a good point. We do have some code that already strips out some sensitive fields ([here](https://github.com/Expensify/Expensify.cash/blob/b7ba89bc9d762e13d344a300237ca7ea0ff5d78b/src/libs/API.js#L90-L95)) so maybe we could move that into a utility at some point so it can be reused.